### PR TITLE
Feature/usage

### PR DESCRIPTION
**Pull request by ankush-jain-akto**

### CODE REVIEW

File: apps/dashboard/src/main/java/com/akto/action/observe/Utils.java [hunk 1]
```diff
@@ -39,6 +40,39 @@ public static List<BasicDBObject> fetchEndpointsInCollectionUsingHost(int apiCol
...
+                .append("method", "$method");
+
+        Bson projections = Projections.fields(
+                Projections.include("timestamp", "lastSeen", "apiCollectionId", "url", "method"));
+
+        pipeline.add(Aggregates.project(projections));
+        pipeline.add(Aggregates.group(groupedId,
+                Accumulators.max(_START_TS, "$timestamp"),
+                Accumulators.max(_LAST_SEEN_TS, "$lastSeen")));
+        pipeline.add(Aggregates.match(filters));
```
Comment (avneesh-akto) [inline apps/dashboard/src/main/java/com/akto/action/observe/Utils.java:61]: Match should come earlier (preferably first). This reduces query time by large amount

Comment (ankush-jain-akto) [reply apps/dashboard/src/main/java/com/akto/action/observe/Utils.java:61]: @notshivansh ^^

File: apps/dashboard/src/main/java/com/akto/action/observe/Utils.java [hunk 2]
```diff
@@ -39,6 +40,39 @@ public static List<BasicDBObject> fetchEndpointsInCollectionUsingHost(int apiCol
...
+        pipeline.add(Aggregates.limit(LARGE_LIMIT));
```
[the remark this answers is not in the document]
Comment (ankush-jain-akto) [reply apps/dashboard/src/main/java/com/akto/action/observe/Utils.java:62]: @notshivansh ^^

File: apps/dashboard/src/main/java/com/akto/action/observe/Utils.java (hunk 1 above)
Comment (notshivansh) [reply apps/dashboard/src/main/java/com/akto/action/observe/Utils.java:58]: @avneesh-akto fixed.

**Pull request merged**